src/datasets/arrow_dataset.py

```diff
@@ -2175,6 +2178,50 @@ def generator():
         logger.info(f"Finished writing TFRecord to {filename}")
         self = None  # delete the dataset reference used by tf_dataset
 
+    def _write_csv(self, file_obj: BinaryIO, batch_size: int, **to_csv_kwargs) -> int:
+        """
+        Writes the pyarrow table as CSV to a binary file handle.
+        Caller is responsible for opening and closing the handle.
+        """
+        written = 0
+        to_csv_kwargs.pop("header", None)
```

Comment:
I think we should allow the header parameter.
something like `head=header if offset == 0 else False`